databusclient/client.py

```
@@ -205,6 +205,56 @@ def create_distribution(
 
     return f"{url}|{meta_string}"
 
+def create_distributions_from_metadata(metadata):
+    distributions = []
+    counter = 0
+    for entry in metadata:
+        filename = entry["filename"]
+        checksum = entry["checksum"]
+        size = entry["size"]
+        if not isinstance(size, int) or size <= 0:
+            raise ValueError(f"Invalid size for {filename}: expected positive integer, got {size}")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)